imports/plugins/included/product-variant/components/customer/productGridItem.js

```diff
@@ -0,0 +1,191 @@
+import React, { Component } from "react";
+import PropTypes from "prop-types";
+import classnames from "classnames";
+import { formatPriceString, Router } from "/client/api";
+import { Components } from "@reactioncommerce/reaction-components";
+
+class ProductGridItem extends Component {
+  static propTypes = {
+    isSearch: PropTypes.bool,
+    position: PropTypes.object,
+    product: PropTypes.object
+  }
+
+  // get product detail page URL
+  get productURL() {
+    const { product: { handle } } = this.props;
+    return Router.pathFor("product", {
+      hash: {
+        handle
+      }
+    });
+  }
+
+  // get weight class name
+  get weightClass() {
+    const { weight } = this.props.position || { weight: 0 };
+    switch (weight) {
+      case 1:
+        return "product-medium";
+      case 2:
+        return "product-large";
+      default:
+        return "product-small";
+    }
+  }
+
+  // get notice class names
+  get noticeClassNames() {
+    const { product: { isSoldOut, isLowQuantity, isBackorder } } = this.props;
+    return classnames({
+      "badge": (isSoldOut || isLowQuantity),
+      "variant-qty-sold-out": (isSoldOut || (isSoldOut && isBackorder)),
+      "badge-danger": (isSoldOut && !isBackorder),
+      "badge-low-inv-warning": (isLowQuantity && !isSoldOut)
+    });
+  }
+
+  // get product item class names
+  get productClassNames() {
+    const { position } = this.props;
+    return classnames({
+      "product-grid-item": true,
+      [this.weightClass]: true,
+      "pinned": position.pinned
+    });
+  }
+
+  // handle click event
+  handleClick = (event) => {
+    event.preventDefault();
+    Router.go(this.productURL);
+  }
+
+  // notice
+  renderNotices() {
+    const { product: { isSoldOut, isLowQuantity, isBackorder } } = this.props;
+    const noticeContent = { classNames: this.noticeClassNames };
+
+    if (isSoldOut) {
+      if (isBackorder) {
+        noticeContent.defaultValue = "Backorder";
+        noticeContent.i18nKey = "productDetail.backOrder";
+      } else {
+        noticeContent.defaultValue = "Sold Out!";
+        noticeContent.i18nKey = "productDetail.soldOut";
+      }
+    } else if (isLowQuantity) {
+      noticeContent.defaultValue = "Limited Supply";
+      noticeContent.i18nKey = "productDetail.limitedSupply";
+    }
+
+    return (
+      <div className="grid-alerts">
+        <div className="product-grid-badges">
+          <span className={noticeContent.classNames}>
+            <Components.Translation defaultValue={noticeContent.defaultValue} i18nKey={noticeContent.i18nKey} />
+          </span>
+        </div>
+      </div>
+    );
+  }
+
+  // render product image
+  renderMedia() {
+    const { product } = this.props;
+    const MEDIA_PLACEHOLDER = "/resources/placeholder.gif";
+    const { large } = (Array.isArray(product.media) && product.media[0]) || { large: MEDIA_PLACEHOLDER };
```

Comment:
`large` seems like a strange const name. Not a big deal at all, just seemed odd reading through it. WHy not just `media` or `img`?

Reply:
`large` is one of the sizes of the media object. `small`, `medium`, 'large`, 'image', `thumbnail`